### PR TITLE
devtools package: fix switchToFrame function

### LINKED ISSUE #5345: devtools protocol: browser.switchToFrame(null) causes a TypeError
**Environment (please complete the following information):**
 - **WebdriverIO version:** 6.1.6
 - **Mode:** WDIO Testrunner
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 12.16.1
 - **NPM version:** 6.13.4
 - **Browser name and version:** Chrome 81
 - **Platform name and version:** macOS 10.15.4
 - **Additional wdio packages used (if applicable):** 

**Config of WebdriverIO**
`automationProtocol: 'devtools'`

**Describe the bug**
Once switched to iframe, and then switch back to parent frame using `browser.switchToFrame(null)`. It causes a TypeError and stop tests.

**To Reproduce**
```javascript
it('Switch to Frame', () => {
  browser.url('https://takeya0x86.github.io/playground/0502_base.html');
  expect($('#paragraph-1')).toHaveText('paragraph 1 - base page');
  browser.switchToFrame($('iframe[name="iframe-1"]'));
  expect($('#paragraph-2')).toHaveText('paragraph 2 - iframe page');
  browser.switchToFrame(null); // => Error!!
  expect($('#paragraph-3')).toHaveText('paragraph 3 - base page');
});
```

**Expected behavior**
not any errors.

**Log**
https://gist.github.com/takeya0x86/f1e9225c2884bffcff10d2227edcb916

**Additional context**
When `automationProtocol: 'webdriver'`, it works fine.

### PR DESCRIPTION
**Pull request by takeya0x86**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

fixes #5345

Add tests
* add switchToFrame(number) test to e2e protocol test suite.
* add switchToFrame(null) test to e2e protocol test suite.

Fix codes
* fix switchToFrame(number) to apply appropriate this value.
* fix condition checking order to avoid reference null value.
* fix val to avoid infinite loop.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/devtools/src/commands/switchToFrame.js [hunk 1]
```diff
@@ -32,7 +44,7 @@ export default async function switchToFrame ({ id }) {
          * `page` has `frames` method while `frame` has `childFrames` method
          */
         let getFrames = page.frames || page.childFrames
-        const childFrames = await getFrames()
+        const childFrames = await getFrames.apply(page)
```
Comment (erwinheitzman) [inline packages/devtools/src/commands/switchToFrame.js:47]: Why not do getFrames(page)?

Comment (christian-bromann) [reply packages/devtools/src/commands/switchToFrame.js:47]: The method doesn't take arguments. @takeya0x86 is correct here we call the method without its scope.

**Comment by christian-bromann** [comment]
Build is currently broken due to #5369 .. let's wait until it gets merged so we can rebase. Thank you @takeya0x86 for providing a fix!

**Comment by christian-bromann** [comment]
I will go ahead and merge this so I can include it in the next release.

**Pull request merged**